app/javascript/bundles/Map/components/RestaurantMapMarker.jsx

```diff
@@ -0,0 +1,72 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import cx from 'classnames';
+import RestaurantCard from "../../RestaurantsApp/components/RestaurantCard";
+import ReactTooltip from 'react-tooltip';
+
+export default class RestaurantMapMarker extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isHovered: false,
+        };
+
+        this.onMouseEnter = this.onMouseEnter.bind(this);
+        this.onMouseLeave = this.onMouseLeave.bind(this);
+    }
+
+    static propTypes = {
+        text: PropTypes.string,
+        restaurant: PropTypes.object.isRequired,
+        onRestaurantsChange: PropTypes.func.isRequired,
+        isSelected: PropTypes.bool,
+    };
+
+    static defaultProps = {
+        text: "",
+        isSelected: false,
+    };
+
+    onMouseEnter() {
+        this.setState({isHovered: true});
+    }
+
+    onMouseLeave() {
+        this.setState({isHovered: false});
+    }
+
+    render() {
+        const markerClass = cx('map-marker', {
+            'selected': this.props.isSelected,
+        });
```

Comment:
😄

Reply:
No need for quotes around `selected`, since it's a valid key name.
You would need them if it contained non-word characters, say, `foo-selected`.